internal/resources/resource_volume.go

```diff
@@ -211,6 +213,22 @@ func volumeSchema() map[string]*schema.Schema {
 			Computed:    true,
 			Description: "The number of active exports for this volume",
 		},
+
+		vDataCenterName: {
+			Type:        schema.TypeString,
+			Required:    false,
+			Optional:    false,
+			Computed:    true,
+			Description: "Name of the data center where the volume is created on the storage array.",
+		},
+
+		vStoragePoolName: {
+			Type:        schema.TypeString,
+			Required:    false,
+			Optional:    false,
```

Comment:
shouldn't one of these be true?

Reply:
There is already a `vStoragePool` attribute present

Reply:
Looks like vStoragePool is not being populated (in the translation layer). Wonder how it's being used by the client (BMaaS UI).

Reply:
Hi @karthik-radakrishnan and @dbozzato81, I realized that vStoragePool is getting populated using /available-resources API. And if I remove StoragePoolName then we are back to the same problem where vStoragePool can have a specific data to a particular pod location. And same is true with DataCanterName. I will keep both of them as these are read only parameters. Let me know if there are any concerns.

Reply:
From the vStoragePool definition and looking at the code, it seams it is only used for Volume create i.e. it is not getting populated by the provider by it is given by the user in case vStoragePoolID is not. 
```golang
		vStoragePool: {
			Type:        schema.TypeString,
			Required:    false,
			Optional:    true,
			Description: "The storage pool of the volume to be created.",
		},
```
So why we can't use it like this?...
```golang
		vStoragePool: {
			Type:        schema.TypeString,
			Optional:    true,
                        Computed:  true,
			Description: "The storage pool of the volume to be created.",
		},
```
Another question, are we sure that vStoragePoolID is computed and returned correctly for external volumes?

Reply:
Hi @dbozzato81, vStoragePoolID is assigned to externally managed volumes when we create them in the DB from DCC. This poolID is assigned based on the array volume was created on.

I have explained above for the vStoragePool - This is fetched based on the vStoragePoolID and available-resources response. And then it is used to create Volume. 

vStoragePoolName and vDataCenterName are read only fields and if we don't want them to have in terraform then we can discard this PR and make vStoragePool **Computed** as **true**